nucliadb_vectors/src/data_types/vector.rs

```diff
@@ -41,6 +41,54 @@ pub fn encode_vector(vec: &[Unit]) -> Vec<u8> {
         .fold(encode_length(vec![], vec), encode_unit)
 }
 
+#[inline]
+#[allow(unused)]
+pub fn simd_consine_similarity(mut x: &[u8], mut y: &[u8]) -> Dist {
+    use wide::f32x8;
+    const N: usize = 8;
+    let mut buff_x = [0; 8];
+    let mut buff_y = [0; 8];
+    x.read_exact(&mut buff_x).unwrap();
+    y.read_exact(&mut buff_y).unwrap();
+    let len_x = Len::from_le_bytes(buff_x);
+    let len_y = Len::from_le_bytes(buff_y);
+    assert_eq!(len_x, len_y);
+    let len = len_x;
+    let mut buff_x = [0; 4];
```

Comment:
is it 4 lane or 8 lane?

Reply:
In that case is 4 bytes, is the buffer used to read the f32 values